pandas/io/json/_normalize.py

```diff
@@ -262,6 +262,11 @@ def _pull_records(js: Dict[str, Any], spec: Union[List, str]) -> List:
     if isinstance(data, list) and not data:
         return DataFrame()
 
+    if isinstance(data, abc.Iterator):
```

Comment:
can you make these if/elif (all 3 conditions)

Reply:
Done